compiler/src/iree/compiler/Codegen/Common/EncodingUtils.cpp

```diff
@@ -7,20 +7,56 @@
 #include "iree/compiler/Codegen/Common/EncodingUtils.h"
 #include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenTypes.h"
 #include "iree/compiler/Codegen/Dialect/Codegen/Utils/Utils.h"
+#include "iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.h"
 #include "iree/compiler/Dialect/Encoding/IR/EncodingTypes.h"
 #include "iree/compiler/Dialect/TensorExt/IR/TensorExtTypes.h"
+#include "llvm/Support/Debug.h"
 #include "mlir/Dialect/Tensor/IR/Tensor.h"
 #include "mlir/Dialect/Utils/IndexingUtils.h"
 #include "mlir/IR/BuiltinAttributes.h"
 
 #include <optional>
 
+#define DEBUG_TYPE "iree-codegen-encoding-utils"
+
 namespace mlir::iree_compiler {
 
 using IREE::Codegen::MaterializeEncodingInfo;
 using IREE::Encoding::EncodingAttr;
 using IREE::Encoding::getEncodingAttr;
 using IREE::Encoding::getEncodingContractionDims;
+using IREE::Encoding::PadEncodingLayoutAttr;
+using IREE::Encoding::SerializableEncodingAttrInterface;
+
+// Returns the layout as a SerializableEncodingAttrInterface, or nullptr if this
+// is not the only layout or if there's no encoding at all.
+static SerializableEncodingAttrInterface
+getSerializableEncodingAttr(Attribute layoutAttr, RankedTensorType type) {
```

Comment:
Do we pass in layoutAttr as `LayoutAttrInterface` type? It reduces the confusion because we have LayoutAttr and EncodingLayoutResolverAttr; it is only used by the type converters.

(It's my fault that I can't find better terminologies.)

Reply:
Yes, indeed, updated it to `LayoutAttrInterface`.